src/encoded/types/analysis.py

```diff
@@ -0,0 +1,141 @@
+        "type": "array",
+        "notSubmittable": True,
+        "items": {
+            "type": "string"
+        }
+    })
+    def assemblies(self, request, files):
+        assemblies = set()
+        for f in files:
+            file_object = request.embed(f, '@@object?skip_calculated=true')
+            if 'assembly' in file_object:
+                assemblies.add(file_object['assembly'])
+        return sorted(assemblies)
+
+    @calculated_property(schema={
+        "title": "Genome Annotations",
+        "description": "A list of genome annotations on which this analysis is performed.",
+        "comment": "Do not submit. This field is calculated from files in this analysis.",
+        "type": "array",
+        "notSubmittable": True,
+        "items": {
+            "type": "string"
+        }
+    })
+    def genome_annotations(self, request, files):
+        genome_annotations = set()
+        for f in files:
+            file_object = request.embed(f, '@@object?skip_calculated=true')
+            if 'genome_annotation' in file_object:
+                genome_annotations.add(file_object['genome_annotation'])
+        return sorted(genome_annotations)
+
+    @calculated_property(
+        define=True,
+        schema={
+            "title": "Pipelines",
+            "description": "A list of pipelines used to generate this analysis.",
+            "comment": "Do not submit. This field is calculated from files in this analysis.",
+            "type": "array",
+            "notSubmittable": True,
+            "items": {
+                "type": "string",
+                "linkTo": "Pipeline"
+            }
+        }
+    )
+    def pipelines(self, request, files):
+        pipelines = set()
+        for f in files:
+            file_object = request.embed(
+                f,
+                '@@object_with_select_calculated_properties?field=analysis_step_version'
+            )
+            if 'analysis_step_version' in file_object:
+                analysis_step = request.embed(
+                    file_object['analysis_step_version'],
+                    '@@object?skip_calculated=true'
+                )['analysis_step']
+                pipelines |= set(
+                    request.embed(
+                        analysis_step,
+                        '@@object_with_select_calculated_properties?field=pipelines'
+                    ).get('pipelines', [])
+                )
+        return sorted(pipelines)
+
+    @calculated_property(schema={
+        "title": "Pipeline awards",
+        "description": "A list of award bioproject phases to which pipelines "
+                       "that used to generate this analysis belong to.",
+        "comment": "Do not submit. This field is calculated from files in this analysis.",
+        "type": "array",
+        "notSubmittable": True,
+        "items": {
+            "type": "string"
+        }
+    })
+    def pipeline_award_rfas(self, request, pipelines=[]):
```

Comment:
I suspect there may be a "valid" corner case where awards and labs have multiple values but I am not sure.  If there isn't one -- see comment on "assemblies"